### PR TITLE
Fixed mgmt, support convenience API for publicNetworkAccess

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#39310](https://github.com/Azure/azure-sdk-for-java/issues/39310)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java [hunk 1]
```diff
@@ -203,6 +203,13 @@ public interface WebAppBase extends HasName, GroupableResource<AppServiceManager
     /** @return an Observable streaming all logs */
     Flux<String> streamAllLogsAsync();
 
+    /**
+     * Whether the web app can be accessed from public network.
+     *
+     * @return whether the web app can be accessed from public network.
+     */
+    boolean isPublicNetworkAccessEnabled();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:211]: I think in the sample PR, the getter is
```
PublicNetworkAccess publicNetworkAccess()
```

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java [hunk 2]
```diff
@@ -71,6 +71,7 @@
 import com.azure.resourcemanager.resources.fluentcore.model.Indexable;
 import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
 import com.azure.resourcemanager.appservice.fluent.models.RemotePrivateEndpointConnectionArmResourceInner;
+import com.azure.resourcemanager.storage.models.PublicNetworkAccess;
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java:74]: Not storage.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 3]
```diff
@@ -124,6 +124,13 @@ public interface KubernetesCluster
      */
     String agentPoolResourceGroup();
 
+    /**
+     * Whether the kubernetes cluster can be accessed from public network.
+     *
+     * @return whether the kubernetes cluster can be accessed from public network.
+     */
+    boolean isPublicNetworkAccessEnabled();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:132]: I think I made it clear in the sample PR, the getter is
```
PublicNetworkAccess publicNetworkAccess()
```
If the module has such class.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java [hunk 4]
```diff
@@ -940,6 +947,19 @@ interface WithNetworkAccess<FluentT> {
              * @return the next stage of the definition
              */
             WithCreate<FluentT> withAccessRule(IpSecurityRestriction ipSecurityRule);
+
+            /**
+             * Enables public network access for the web app.
+             *
+             * @return the next stage of the definition
+             */
+            WithCreate<FluentT> enablePublicNetworkAccess();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:956]: No `enable` in create flow.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/WebAppsTests.java [hunk 5]
```diff
@@ -284,4 +284,64 @@ public void canUpdateIpRestriction() {
...
+                .define(webappName1)
+                .withRegion(Region.US_WEST)
+                .withExistingResourceGroup(rgName1)
+                .withNewWindowsPlan(appServicePlanName1, PricingTier.BASIC_B1)
+                .enablePublicNetworkAccess()
+                .withRemoteDebuggingEnabled(RemoteVisualStudioVersion.VS2019)
+                .create();
+        webApp.refresh();
+        Assertions.assertTrue(webApp.isPublicNetworkAccessEnabled());
+    }
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/test/java/com/azure/resourcemanager/appservice/WebAppsTests.java:304]: Do not need this. Enable is the default.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java [hunk 6]
```diff
@@ -1835,6 +1835,29 @@ public FluentImplT withoutIpAddressRangeAccess(String ipAddressCidr) {
         return (FluentImplT) this;
     }
 
+    @Override
+    @SuppressWarnings("unchecked")
+    public FluentImplT enablePublicNetworkAccess() {
+        this.ensureIpSecurityRestrictions();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java:1841]: Why we need to call `ensureIpSecurityRestrictions()`? Anything here related to IP?

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java [hunk 7]
```diff
@@ -203,6 +203,13 @@ public interface WebAppBase extends HasName, GroupableResource<AppServiceManager
     /** @return an Observable streaming all logs */
     Flux<String> streamAllLogsAsync();
 
+    /**
+     * Whether the web app can be accessed from public network.
+     *
+     * @return whether the web app can be accessed from public network.
+     */
+    String publicNetworkAccess();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:211]: @XiaofeiCao Should we add a `PublicNetworkAccess` class to the module, to keep the consistency?

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java [hunk 8]
```diff
@@ -1835,6 +1835,29 @@ public FluentImplT withoutIpAddressRangeAccess(String ipAddressCidr) {
...
+    @Override
+    @SuppressWarnings("unchecked")
+    public FluentImplT disablePublicNetworkAccess() {
+        this.ensureIpSecurityRestrictions();
+        this.siteConfig.withPublicNetworkAccess("Disabled");
+        return (FluentImplT) this;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java:1855]: why suppress?

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java [hunk 9]
```diff
@@ -299,6 +301,11 @@ public String agentPoolResourceGroup() {
         return innerModel().nodeResourceGroup();
     }
 
+    @Override
+    public PublicNetworkAccess publicNetworkAccess() {
+        return Objects.isNull(this.innerModel().publicNetworkAccess()) ? null : this.innerModel().publicNetworkAccess();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java:306]: What's difference from just `return this.innerModel().publicNetworkAccess()`?

Same on other instances.

File: sdk/resourcemanager/azure-resourcemanager-cosmos/src/main/java/com/azure/resourcemanager/cosmos/models/CosmosDBAccount.java [hunk 10]
```diff
@@ -613,5 +632,21 @@ PrivateEndpointConnection.UpdateDefinitionStages.Blank<WithOptionals> defineNewP
...
+             *
+             * @return the next stage of the update
+             */
+            Update enablePublicNetworkAccess();
+            /**
+             * Disables public network access for the cosmos db.
+             *
+             * @return the next stage of the update
+             */
+            Update disablePublicNetworkAccess();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-cosmos/src/main/java/com/azure/resourcemanager/cosmos/models/CosmosDBAccount.java:649]: Doc, follow other interfaces, "CosmosDB account", not "cosmos db".

File: sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java [hunk 11]
```diff
@@ -714,4 +715,35 @@ public void updateIdentityFromNoneToSystemUserAssigned() {
...
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withSystemAssignedManagedServiceIdentity()
+            .disablePublicNetworkAccess()
+            .create();
+        Assertions.assertEquals(PublicNetworkAccess.DISABLED, storageAccount.publicNetworkAccess());
+    }
+
+    @Test
+    public void updatePublicNetworkAccess() {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/test/java/com/azure/resourcemanager/storage/StorageAccountOperationsTests.java:734]: canUpdatePublicNetworkAccess

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java [hunk 12]
```diff
@@ -987,8 +999,22 @@ interface WithBlobAccess {
             Update disableSharedKeyAccess();
         }
 
-        /** The stage of storage account update allowing to configure network access. */
+        /** The stage of storage update allowing to configure network access. */
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java:1002]: Why this change? It is "storage account".

File: sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java [hunk 13]
```diff
@@ -987,8 +999,22 @@ interface WithBlobAccess {
...
         interface WithNetworkAccess {
+            /**
+             * Enables public network access for the storage.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java:1005]: Same here, storage account.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java (hunk 7 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:211]: Seems this is the only module that not having it.

I'd prefer add one. Just copy it from another module, update doc, remove line 3 `// Code generated by Microsoft (R) AutoRest Code Generator.`.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:211]: Please add a `PublicNetworkAccess` class to the module.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java [hunk 14]
```diff
@@ -626,4 +627,95 @@ public void testUpdateManagedClusterSkuAndKubernetesSupportPlan() {
...
+    public void canCreateKubernetesClusterWithDisablePublicNetworkAccess() {
+        String aksName = generateRandomResourceName("aks", 15);
+        String dnsPrefix = generateRandomResourceName("dns", 10);
+        String agentPoolName = generateRandomResourceName("ap0", 10);
+        String agentPoolName1 = generateRandomResourceName("ap1", 10);
+
+        String agentPoolResourceGroupName = generateRandomResourceName("pool", 15);
+
+        // create
+        KubernetesCluster kubernetesCluster =
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java:641]: simplify your resource in test, you don't need all these configure.

a simple one would do
https://github.com/Azure/azure-sdk-for-java/blob/5b6d47ec83296d5a2e4ac539740eea5f3b008791/sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java#L365-L379

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java (hunk 7 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/WebAppBase.java:211]: Yeah, I think we should.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/PublicNetworkAccess.java [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * Static value Disabled for PublicNetworkAccess.
+     */
+    public static final PublicNetworkAccess DISABLED = fromString("Disabled");
+
+    /**
+     * Static value SecuredByPerimeter for PublicNetworkAccess.
+     */
+    public static final PublicNetworkAccess SECURED_BY_PERIMETER = fromString("SecuredByPerimeter");
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/models/PublicNetworkAccess.java:28]: remove this

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java [hunk 16]
```diff
@@ -1835,6 +1836,33 @@ public FluentImplT withoutIpAddressRangeAccess(String ipAddressCidr) {
...
+        }
+        this.siteConfig.withPublicNetworkAccess("Disabled");
+        return (FluentImplT) this;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public PublicNetworkAccess publicNetworkAccess() {
+        return Objects.isNull(innerModel().publicNetworkAccess())
+            && innerModel().publicNetworkAccess().trim().isEmpty() ? null : PublicNetworkAccess.fromString(innerModel().publicNetworkAccess());
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java:1863]: No need to trim or check empty, just do `PublicNetworkAccess.fromString`, if not null.

File: sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java (hunk 8 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java:1860]: I still not see what your resolution on this.

**Review by weidongxu-microsoft — COMMENTED** [review]
LGTM.

Please add the changelog.

For WebApp, please do a double check, that after you disable public network via SDK, the Portal UI (not JSON template) or CLI would show this as "disabled". Confirm by a screenshot please.
Reason is that WebApp (Site) has 2 model that have similar structure, but only one of them takes effect.

**Comment by v-hongli1** [comment]
Create web app with `disablePublicNetworkAccess()`
image

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-cosmos/src/main/java/com/azure/resourcemanager/cosmos/models/CosmosDBAccount.java [hunk 17]
```diff
@@ -36,6 +36,13 @@ public interface CosmosDBAccount
     /** @return the offer type for the CosmosDB database account */
     DatabaseAccountOfferType databaseAccountOfferType();
 
+    /**
+     * Whether the CosmosD account can be accessed from public network.
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-cosmos/src/main/java/com/azure/resourcemanager/cosmos/models/CosmosDBAccount.java:40]: typo, `CosmosDB`.

**Review by XiaofeiCao — APPROVED** [review]
